File: Strings/1_10/6Solution.py
def calPoints(ops):
        i = j = 0
        while j < len(ops):
            if ops[j] == "C":
                i -= 2
            elif ops[j] == "D":
                ops[i] = int(ops[i-1])*2
            elif ops[j] == "+":
                ops[i] = int(ops[i-1]) + int(ops[i-2])
            else:
                ops[i] = int(ops[j])
            i += 1
            j += 1
        return sum(ops[:i])
ops=[ x.strip() for x in input().split(",")]
print(calPoints(ops))

File: Strings/1_10/6Solution.java
class Solution {
    public int calPoints(String[] ops) {
        int[] stack=new int[ops.length];
        int index=-1;
        for(int i=0;i<ops.length;i++){
            if(ops[i].equals("D")){
                stack[++index]=stack[index-1]*2;
            }else if(ops[i].equals("C")){
                stack[index--]=0;
            }else if(ops[i].equals("+")){
                stack[++index]=stack[index-1]+stack[index-2];
            }else{
                stack[++index]=Integer.parseInt(ops[i]);
            }
        }
        int sum=0;
        for(int i=0;i<=index;i++){
            sum+=stack[i];
        }
        return sum;
    }
}
